zero/src/rpc/jumpdest.rs

```diff
@@ -0,0 +1,429 @@
+use core::default::Default;
+use core::option::Option::None;
+use core::str::FromStr as _;
+use std::collections::HashMap;
+use std::ops::Not as _;
+use std::time::Duration;
+
+use ::compat::Compat;
+use __compat_primitive_types::H160;
+use __compat_primitive_types::H256;
+use alloy::eips::BlockNumberOrTag;
+use alloy::primitives::Address;
+use alloy::providers::ext::DebugApi;
+use alloy::providers::Provider;
+use alloy::rpc::types::eth::Transaction;
+use alloy::rpc::types::trace::geth::{
+    GethDebugTracingOptions, GethDefaultTracingOptions, GethTrace, StructLog, TraceResult,
+};
+use alloy::transports::Transport;
+use alloy_primitives::{TxHash, U256};
+use anyhow::bail;
+use anyhow::ensure;
+use evm_arithmetization::jumpdest::JumpDestTableWitness;
+use keccak_hash::keccak;
+use tokio::time::timeout;
+use trace_decoder::is_precompile;
+use trace_decoder::ContractCodeUsage;
+use trace_decoder::TxnTrace;
+use tracing::{trace, warn};
+
+/// The maximum time we are willing to wait for a block's structlogs  before
+/// failing over to simulating the JumpDest analysis.
+const TIMEOUT_LIMIT: Duration = Duration::from_secs(60);
```

Comment:
would ideally be up to the user to specify a timeout value

Reply:
`--timeout` is now supported.